File: benchmarks/javascript/map_lookup.js
const n = Number(process.argv[2] ?? 200000);
const index = new Map();
for (let i = 0; i < n; i++) {
    index.set("k" + i, i);
}
let total = 0;
for (let i = 0; i < n; i++) {
    total = total + (index.get("k" + i) % 1000);
}
console.log(total, index.size);

File: benchmarks/python/map_lookup.py
import sys

def main():
    n = int(sys.argv[1]) if len(sys.argv) > 1 else 200000
    index = {}
    for i in range(n):
        index["k" + str(i)] = i
    total = 0
    for i in range(n):
        total = total + index["k" + str(i)] % 1000
    print(total, len(index))

main()
